### PR TITLE
Hairbows - 14 versions for people and monkeys

### PR DESCRIPTION
**Pull request by jan-antilles**
[Sprites][Add to Wiki][Feature]

## About the PR 
This adds 14 new hairbows to the game! They are available in the Clothing Booth, except for one Flashy Hairbow that can be found in eggs. They fit all the mutant races, sit nicely on AIs and Cyborgs and Ghostdrones, and even have slightly smaller but still adorable versions for monkeys!

hairbowspreview

## Why's this needed? 

image

Bows are cute and customization is good!

## Changelog

```changelog
(u)Jan.antilles
(*)Added hairbows! Find them in the Clothing Booth!
```

### CODE REVIEW

File: code/modules/vending/clothingbooth_items.dm [hunk 1]
```diff
@@ -46,6 +46,90 @@
 	category = "Accessories"
 	cost = PAY_TRADESMAN/3
 
+/datum/clothingbooth_item/hbowmagenta
+	name = "Magenta Hair Bow"
+	path = /obj/item/clothing/head/hairbow/magenta
+	slot = SLOT_HEAD
+	category = "Accessories"
+	cost = PAY_TRADESMAN/3
```
Comment (MarkNais) [inline code/modules/vending/clothingbooth_items.dm:54]: Thought:
I imagine this was annoying to type out. A medium-sized #define could be used in the future to save on some typing.
First example that came to mind: (it's a doozy of an example)
https://github.com/goonstation/goonstation/blob/f173e5ebf7fe6952444e8f81b88811f686b696dc/code/modules/power/cablecolors.dm#L1

**Review by MarkNais — APPROVED** [review]

File: code/modules/vending/clothingbooth_items.dm (hunk 1 above)
Comment (jan-antilles) [reply code/modules/vending/clothingbooth_items.dm:54]: Yeah, that's definitely something I'll keep in mind for the future. This PR I got to learn about the magical world of child objects, so hopefully next time I can try that!

File: code/modules/vending/clothingbooth_items.dm [hunk 2]
```diff
@@ -46,6 +46,90 @@
 	category = "Accessories"
 	cost = PAY_TRADESMAN/3
 
+/datum/clothingbooth_item/hbowmagenta
```
Comment (ZeWaka) [inline code/modules/vending/clothingbooth_items.dm:49]: why are these all distinct subtypes instead of being commonly inherited?

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/vending/clothingbooth_items.dm (hunk 2 above)
Comment (Sovexe) [reply code/modules/vending/clothingbooth_items.dm:49]: https://github.com/goonstation/goonstation/blob/987aee5b61d6f7327da2f4198cf8ee173f0f11e9/code/modules/vending/clothingbooth.dm#L8

You will need to change this line as well to support ZeWakas request to abstract and subtype the bows

Comment (Sovexe) [reply code/modules/vending/clothingbooth_items.dm:49]: #5480

File: code/modules/vending/clothingbooth_items.dm [hunk 3]
```diff
@@ -46,6 +46,90 @@
...
+	category = "Accessories"
+	cost = PAY_TRADESMAN/3
+
+/datum/clothingbooth_item/hbowrainbow
+	name = "Rainbow Hair Bow"
+	path = /obj/item/clothing/head/hairbow/rainbow
+	slot = SLOT_HEAD
+	category = "Accessories"
+	cost = PAY_TRADESMAN
+
```
Comment (Sovexe) [inline code/modules/vending/clothingbooth_items.dm:132]:
```suggestion
ABSTRACT_TYPE(/datum/clothingbooth_item/accessory/hbow/)
/datum/clothingbooth_item/accessory/hbow/
	name = "Hair Bow"
	slot = SLOT_HEAD

	magenta
		name = "Magenta Hair Bow"
		path = /obj/item/clothing/head/hairbow/magenta

	pink
		name = "Pink Hair Bow"
		path = /obj/item/clothing/head/hairbow/pink

	red
		name = "Red Hair Bow"
		path = /obj/item/clothing/head/hairbow/red

	gold
		name = "Gold Hair Bow"
		path = /obj/item/clothing/head/hairbow/gold

	green
		name = "Green Hair Bow"
		path = /obj/item/clothing/head/hairbow/green

	mint
		name = "Mint Hair Bow"
		path = /obj/item/clothing/head/hairbow/mint

	blue
		name = "Blue Hair Bow"
		path = /obj/item/clothing/head/hairbow/blue

	navy
		name = "Navy Hair Bow"
		path = /obj/item/clothing/head/hairbow/navy

	shinyblack
		name = "Shiny Black Hair Bow"
		path = /obj/item/clothing/head/hairbow/shinyblack

	matteblack
		name = "Matte Black Hair Bow"
		path = /obj/item/clothing/head/hairbow/matteblack

	white
		name = "White Hair Bow"
		path = /obj/item/clothing/head/hairbow/white

	rainbow
		name = "Rainbow Hair Bow"
		path = /obj/item/clothing/head/hairbow/rainbow
```

Comment (ZeWaka) [reply code/modules/vending/clothingbooth_items.dm:132]: base will also need updating

Comment (Sovexe) [reply code/modules/vending/clothingbooth_items.dm:132]: I believe a pull from master should already include that

Comment (ZeWaka) [reply code/modules/vending/clothingbooth_items.dm:132]: correct

**Review by Sovexe — APPROVED** [review]

**Pull request merged**